### ISSUE TITLE
fn about anonymous class and unusedformalparameter

### ISSUE THREAD

**Issue by LynnBroe** [pmd/pmd#4064]
**Affects PMD Version:** 

**Rule: unusedformalparameter**

<https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedformalparameter>

**Description: Hi. When the code usa an anonymous class, the PMD should report a warning. I think that it doesn't consider the anonymous class.**

**Code Sample demonstrating the issue:**

```
Object a = new Object(){
    public void foo(int a){
    
    }
};
```

**Expected outcome:**

PMD should report a unusedformalparameter warning.

**Running PMD through:** *CLI*
